### DISCUSSION TITLE
Need code coverage for the kernel components.

### DISCUSSION THREAD

**Discussion by shankarseal** [microsoft/ebpf-for-windows#863] [General]
The following tools can perform code coverage. Buy the licenses are not free.

https://www.bullseye.com/cgi-bin/store
https://www.qt.io/pricing?hsCtaTracking=e985704c-120c-4c8b-8f40-310c5be9904d%7C0cde677c-a91a-463a-958c-1cf4fe5d57ff

**Comment by dthaler** [comment]
BSC asked: can Coverity do this?
